File: python/pplx-setup.py
#!/usr/bin/env python3
# pplx-setup.sh — portiert nach python
# Quelle: shell, OpenClaw@main:scripts/pplx-tools/pplx-setup.sh
# Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

import os
import subprocess
import sys
from pathlib import Path

# One-time (idempotent): make sure the Perplexity VS Code extension daemon can
# find a Chromium. The daemon uses its OWN bundled patchright, which pins a
# specific chromium revision; install exactly that revision.

def main():
    home = Path.home()
    ext_dirs = list(home.glob('.vscode-remote/extensions/nskha.perplexity-vscode-*/dist/node_modules/patchright'))
    
    if not ext_dirs:
        print("[setup] extension patchright not found — is the Perplexity extension installed?")
        return 0
    
    # Sort by version and get the latest
    ext_dirs.sort(key=lambda x: str(x))
    extpr = ext_dirs[-1]
    
    try:
        result = subprocess.run([
            'node', '-e', 
            f"const {{chromium}}=require('{extpr}');console.log(chromium.executablePath())"
        ], capture_output=True, text=True, check=True)
        exp = result.stdout.strip()
    except subprocess.CalledProcessError:
        exp = ""
    
    if exp and Path(exp).is_file() and os.access(exp, os.X_OK):
        print(f"[setup] daemon browser already present: {exp}")
        return 0
    
    print(f"[setup] installing matching chromium for the extension daemon (expected: {exp or 'unknown'})...")
    subprocess.run(['node', str(extpr / 'cli.js'), 'install', 'chromium'], check=True)
    print("[setup] done.")
    return 0

if __name__ == '__main__':
    sys.exit(main())

File: javascript/pplx-setup.js
#!/usr/bin/env node
// pplx-setup.sh — portiert nach javascript
// Quelle: shell, OpenClaw@main:scripts/pplx-tools/pplx-setup.sh
// Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

// One-time (idempotent): make sure the Perplexity VS Code extension daemon can
// find a Chromium. The daemon uses its OWN bundled patchright, which pins a
// specific chromium revision; install exactly that revision.
'use strict';

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

function findExtensionPatchright() {
  const extensionsDir = path.join(process.env.HOME, '.vscode-remote', 'extensions');
  try {
    const files = fs.readdirSync(extensionsDir);
    const patchrightDirs = files
      .filter(file => file.startsWith('nskha.perplexity-vscode-'))
      .map(dir => path.join(extensionsDir, dir, 'dist', 'node_modules', 'patchright'))
      .filter(dir => fs.existsSync(dir))
      .sort((a, b) => a.localeCompare(b, undefined, { numeric: true, sensitivity: 'base' }));
    
    return patchrightDirs.length > 0 ? patchrightDirs[patchrightDirs.length - 1] : null;
  } catch (err) {
    return null;
  }
}

const extpr = findExtensionPatchright();
if (!extpr) {
  console.log('[setup] extension patchright not found — is the Perplexity extension installed?');
  process.exit(0);
}

let exp = '';
try {
  exp = execSync(`node -e "const {chromium}=require('${extpr}');console.log(chromium.executablePath())"`, {
    stdio: ['pipe', 'pipe', 'ignore']
  }).toString().trim();
} catch (err) {
  // Ignore error, exp remains empty
}

if (exp && fs.existsSync(exp) && fs.statSync(exp).isFile()) {
  console.log(`[setup] daemon browser already present: ${exp}`);
  process.exit(0);
}

console.log(`[setup] installing matching chromium for the extension daemon (expected: ${exp || 'unknown'})...`);
execSync(`node "${path.join(extpr, 'cli.js')}" install chromium`, { stdio: 'inherit' });
console.log('[setup] done.');
